### PR TITLE
Add attestation quote in batch poster

### PR DESCRIPTION
**Pull request by Sneh1999**
Closes #[286](https://github.com/EspressoSystems/nitro-espresso-integration/issues/286)

image

### This PR:
It helps us generate attestation quote which will later be used by the contract to do verifications

### CODE REVIEW

File: arbnode/batch_poster.go [hunk 1]
```diff
@@ -1138,15 +1169,58 @@ func (b *BatchPoster) encodeAddBatch(
...
+	}
+
+	// Write the message to "/dev/attestation/user_report_data" in SGX
+	err := os.WriteFile(b.config().UserDataAttestationFile, userDataHash, 0600)
+	if err != nil {
+		return []byte{}, fmt.Errorf("failed to create user report data file: %w", err)
+	}
+
+	// Read the quote from "/dev/attestation/quote" in SGX
+	attestationQuote, err := os.ReadFile(b.config().QuoteFile)
```
Comment (ImJeremyHe) [inline arbnode/batch_poster.go:1215]: Does that mean you get the quote as soon as you write the data to `/dev/attestation/user_report_data`? No need to wait? That's interesting.

File: arbnode/batch_poster.go [hunk 2]
```diff
@@ -182,10 +185,10 @@ type BatchPosterConfig struct {
...
 	ReorgResistanceMargin          time.Duration               `koanf:"reorg-resistance-margin" reload:"hot"`
 	CheckBatchCorrectness          bool                        `koanf:"check-batch-correctness"`
-
-	gasRefunder  common.Address
-	l1BlockBound l1BlockBound
-
+	UserDataAttestationFile        string                      `koanf:"user-data-attestation-file"`
+	QuoteFile                      string                      `koanf:"quote-file"`
+	gasRefunder                    common.Address
+	l1BlockBound                   l1BlockBound
```
Comment (ImJeremyHe) [inline arbnode/batch_poster.go:191]: let's reorder this to make a nicer diff. This helps us in syncing

**Review by ImJeremyHe — APPROVED** [review]
Just some comments

File: arbnode/batch_poster.go (hunk 1 above)
Comment (Sneh1999) [reply arbnode/batch_poster.go:1215]: Docs are [here](https://gramine.readthedocs.io/en/stable/attestation.html#remote-attestation-flows-for-epid-and-dcap) but yes, its pretty interesting. They have a special hardware instruction `EREPORT`. I am guessing as soon as the user closes the `user_report_data` file, they lock the mutex for write for the `quote` file. Other user's probably have to wait before the write operation completes to read. This is my interpretation of how it is probably working but not 100% sure.

**Pull request merged**